runtime/service/src/main/java/org/apache/polaris/service/events/AttributeKey.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+import java.util.Objects;
+
+/**
+ * A type-safe key for event attributes. This allows for strongly-typed attribute access while
+ * maintaining flexibility for custom attributes.
+ *
+ * @param <T> the type of the attribute value
+ */
+public final class AttributeKey<T> {
```

Comment:
Should we add some Jackson annotations here to customize Json serialization? I think this class could be serialized as a string scalar containing just the attribute name, wdyt?